kill.go

```diff
@@ -91,17 +91,31 @@ func kill(containerID, signal string) error {
 		return err
 	}
 
-	// container MUST be running
 	podStatus, err := vc.StatusPod(containerID)
 	if err != nil {
 		return err
 	}
-	if podStatus.State.State != vc.StateRunning {
-		return fmt.Errorf("Container is not running")
-	}
 
 	if len(podStatus.ContainersStatus) != 1 {
-		return fmt.Errorf("ContainerStatus list from PodStatus is wrong, expecting only one ContainerStatus")
+		return fmt.Errorf("ContainerStatus list from PodStatus %s is wrong, expecting only one ContainerStatus", containerID)
+	}
+
+	// container MUST be running
+	if podStatus.ContainersStatus[0].State.State != vc.StateRunning {
+		return fmt.Errorf("Container %s is not running", containerID)
+	}
+
+	// Check status of process
+	running, err := processRunning(podStatus.ContainersStatus[0].PID)
+	if err != nil {
+		return err
+	}
+	if running == false {
+		if err := stopContainer(podStatus); err != nil {
+			return err
+		}
+
+		return fmt.Errorf("Process not running inside container %s", containerID)
```

Comment:
What do you think is racy here ? And about the checks we're doing here, I think we have to do them at that level to ensure we are OCI compliant. Indeed, we can perform some checks from virtcontainers library but it is not OCI compliant, that's why we have to do them in the runtime implementation.